packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/V1_ProtocolToMetaModelPropertyMappingVisitor.ts

```diff
@@ -571,9 +612,74 @@ export class V1_ProtocolToMetaModelPropertyMappingVisitor
     return otherwiseEmbedded;
   }
 
+  visit_XStorePropertyMapping(
+    protocol: V1_XStorePropertyMapping,
+  ): PropertyMapping {
+    assertNonNullable(
+      protocol.property,
+      'XStore property mapping property is missing',
+    );
+    assertNonEmptyString(
+      protocol.property.class,
+      'XStore property mapping property class is missing',
+    );
+    assertNonEmptyString(
+      protocol.property.property,
+      'XStore property mapping property name is missing',
+    );
+    assertNonNullable(
+      protocol.crossExpression,
+      'XStore property mapping cross expression lambda is missing',
+    );
+
+    const xStoreParent = guaranteeNonNullable(
+      this.xStoreParent,
+      'XStoreAssociation expected to be parent of XStore PropertyMapping',
+    );
+    const _association = xStoreParent.association.value;
+    const property = _association.getProperty(protocol.property.property);
+    const sourceSetImplementation = this.allClassMappings.find(
+      (c) => c.id.value === protocol.source,
+    );
+    const targetSetImplementation = this.allClassMappings.find(
+      (c) => c.id.value === protocol.target,
+    );
+    const xStorePropertyMapping = new XStorePropertyMapping(
+      xStoreParent,
+      PropertyExplicitReference.create(property),
+      guaranteeNonNullable(sourceSetImplementation),
+      targetSetImplementation,
+    );
+    xStorePropertyMapping.crossExpression = new RawLambda(
+      protocol.crossExpression.parameters,
+      protocol.crossExpression.body,
+    );
+    return xStorePropertyMapping;
+  }
+
   visit_AggregationAwarePropertyMapping(
     protocol: V1_AggregationAwarePropertyMapping,
   ): PropertyMapping {
-    throw new UnsupportedOperationError();
+    const _class = this.context.resolveClass(protocol.property.class);
```

Comment:
if so, looks like we have just supported aggregation-aware. Should we add a round-trip test for aggregation-aware stuffs as well??

Reply:
Mhm will remove this. The class aggregation aware processing is required for this as well.